### ISSUE TITLE
.header{  SyntaxError: Unexpected token '.' its ssr app stylesheet is creating the issue. please don't close until fix.

### ISSUE THREAD

**Issue by virkse** [webpack-contrib/css-loader#1068]
@evilebottnawi and other community can help me to sort out it. I'll be thank full.
Please help out me, css is not rendering, even i configure css-loader. It is server side rendering, .ejs template used.

package.json

```
{
  "name": "virkse-react-ssr",
  "version": "1.0.0",
  "description": "React server side rendering - complete basic website",
  "main": "index.js",
  "scripts": {
    "dev": "webpack --mode development && concurrently \"webpack --watch\" \"nodemon --exec babel-node src/server.js\""
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "compression": "^1.7.4",
    "ejs": "^3.0.2",
    "express": "^4.17.1",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
    "react-router-dom": "^5.1.2",
    "webpack": "^4.42.1"
  },
  "devDependencies": {
    "@babel/cli": "^7.8.4",
    "@babel/core": "^7.9.0",
    "@babel/node": "^7.8.7",
    "@babel/plugin-proposal-class-properties": "^7.8.3",
    "@babel/polyfill": "^7.8.7",
    "@babel/preset-env": "^7.9.0",
    "@babel/preset-react": "^7.9.4",
    "babel-loader": "^8.1.0",
    "concurrently": "^5.1.0",
    "css-loader": "^3.4.2",
    "mini-css-extract-plugin": "^0.9.0",
    "nodemon": "^2.0.2",
    "style-loader": "^1.1.3",
    "webpack-cli": "^3.3.11"
  }
}
```

**File which has stylesheet imported**

```
import React from 'react'
import './header.css' <------------This is the issue, if i empty the stylesheet then issue resolve.
/** partial import */
import Logo from './partial/Logo'
import HeaderMenues from './partial/HeaderMenues'

class Header extends React.Component {

    render() {
        return (
            <header>
                <div className="header_inner">
                </div>
            </header>
        );
    }
}
export default Header;
```

**My webpack.config.js file**

```
const path = require("path");
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

const config = {
    mode: 'development',
    entry: {
        vendor: ["@babel/polyfill", "react"], // Third party libraries
        index: ["./src/components/entrypoints/application.js"]
        /// Every pages entry point should be mentioned here
    },
    output: {
        path: path.resolve(__dirname, "src", "public"), //destination for bundled output is under ./src/public
        filename: "[name].js" // names of the bundled file will be name of the entry files (mentioned above)
    },
    module: {
        rules: [
            {
                test: /\.(js|jsx)$/,
                use: {
                    loader: "babel-loader", // asks bundler to use babel loader to transpile es2015 code
                    options: {
                        presets: ["@babel/preset-env", "@babel/preset-react"] 
                    }
                },
                exclude: [/node_modules/, /public/]
            },
            {
                //test: /\.(css|scss)$/,
                test: /\.css$/,
                use: [ 
                    {
                        loader: MiniCssExtractPlugin.loader,
                        options: {
                          publicPath: '/style.css',
                        },
                    },
                    'css-loader',
                ]
            },
            {
                test: /\.jpe?g$|\.gif$|\.png$|\.svg$/i,
                loader: 'url-loader?limit=10000'
            },
        ]
    },
    plugins: [
        new MiniCssExtractPlugin({
            filename: '[name].css',
        })
    ],
    resolve: {
        extensions: [".js", ".jsx", ".json", ".wasm", ".mjs", "*"]
    } // If multiple files share the same name but have different extensions, webpack will resolve the one with the extension listed first in the array and skip the rest.
};

module.exports = config;
```

**Comment by virkse** [comment]
**This is the error i getting**

D:\learning\virkse-react-ssr\src\components\common\header.css:1
[1] .ddheader{
[1] ^
[1]
[1] SyntaxError: Unexpected token '.'
[1]     at Module._compile (internal/modules/cjs/loader.js:892:18)
[1]     at Module._extensions..js (internal/modules/cjs/loader.js:973:10)
[1]     at Object.newLoader [as .js] (D:\learning\virkse-react-ssr\node_modules\pirates\lib\index.js:104:7)
[1]     at Module.load (internal/modules/cjs/loader.js:812:32)
[1]     at Function.Module._load (internal/modules/cjs/loader.js:724:14)
[1]     at Module.require (internal/modules/cjs/loader.js:849:19)
[1]     at require (internal/modules/cjs/helpers.js:74:18)
[1]     at Object.<anonymous> (D:\learning\virkse-react-ssr\src\components\common\/Header.js:3:1)
[1]     at Module._compile (internal/modules/cjs/loader.js:956:30)
    at Module._compile (D:\learning\virkse-react-ssr\node_modules\pirates\lib\index.js:99:24)
[1] [nodemon] app crashed - waiting for file changes before starting...

**Comment by evilebottnawi** [comment]
You need `style-loader`, because using without style-loader
```
import './style.css';
```

return `string` and it is not valid js syntax

**Comment by evilebottnawi** [comment]
Please create **minimum** reproducible test repo, I don't know full architecture, and your versions and other information

**Comment by virkse** [comment]
@evilebottnawi  i created here the repo https://github.com/virkse/react-ssr can you please have a look?

**Comment by evilebottnawi** [comment]
It is expected, you don't run webpack on server and node.js doesn't support importing CSS, here good article how to achieve it - https://medium.com/front-end-weekly/adding-a-server-side-rendering-support-for-an-existing-react-application-using-express-and-webpack-5a3d60cf9762, Anyway feel free to create a new issue in webpack docs, it will be great to have an article in official docs how to implement SSR, anyway it is out of scope css-loader, you will faced with same issue in any non JS loader
